src/main/java/uk/gov/hmcts/darts/common/repository/TransformedMediaRepository.java

```diff
@@ -58,4 +59,30 @@ AND upper(tod.status.description) <> 'MARKED FOR DELETION'
         """)
     List<TransformedMediaEntity> findAllDeletableTransformedMedia(OffsetDateTime createdAtOrLastAccessedDateTime);
 
-}
+    @Query("""
+        SELECT tm
+            FROM TransformedMediaEntity tm
+            JOIN tm.mediaRequest media
+            JOIN media.hearing hearing
+            JOIN hearing.courtCase courtCase
+            JOIN hearing.courtroom courtroom
+            JOIN courtroom.courthouse courthouse
+        WHERE
+           (:mediaId IS NULL OR (:mediaId IS NOT NULL AND media.id=:mediaId)) AND
+           (:caseNumber IS NULL  OR (:caseNumber IS NOT NULL AND courtCase.caseNumber=:caseNumber)) AND
+           (:courtHouseDisplayName IS NULL OR (:courtHouseDisplayName IS NOT NULL AND courthouse.displayName like CONCAT('%', :courtHouseDisplayName, '%'))) AND
+           (:hearingDate IS NULL OR (:hearingDate IS NOT NULL AND hearing.hearingDate=:hearingDate )) AND
+           (:owner IS NULL OR (:owner IS NOT NULL AND media.currentOwner.userFullName like CONCAT('%', :owner, '%'))) AND
```

Comment:
how is owner being populated on the screens? i.e. does this need to be a case-insensitive search?

Reply:
Sorry @chrisbellingham-hmcts ? The requirements didnt seem to mention case insensitive but happy to add it